packages/search-service/src/search-mapping/typedef.graphql

```diff
@@ -1,99 +1,154 @@
-scalar DateTime
+type Query {
+  """
+  Fetch active fields and configs and maps
+  """
+  listSearchMap: [SearchMap]
+  """
+  Fetch Specific search map with _id
+  """
+  getSearchMap(id: String!): SearchMap
+  """
+  Fetch Search Map for a given App by appId
+  """
+  getSearchMapsByApp(appId: String!): [SearchMap]
 
-type FieldType {
-    from: String
-    to: String
+  """
+  Trigger search indexing job
+  """
+  triggerSearchMap: String
 }
 
-
-type ApiConfigType {
-    mode: String
-    apiUrl: String
-    query: String
-    param: String
+type Mutation {
+  """
+  Add a new Search Map configuration in database
+  """
+  createSearchMap(appId: String!, searchMap: CreateSearchMapInput!): SearchMap
+  """
+  update the Search Map configuration in database
+  """
+  updateSearchMap(appId: String!, searchMap: UpdateSearchMapInput!): SearchMap
+  """
+  Delete Search Map configuration from database
+  """
+  deleteSearchMap(id: String!): SearchMap
 }
 
-type SearchMapPreferenceType {
-    iconUrl: String,
-    urlTemplate: String
-    urlParams: [String]
-    titleTemplate: String
-    titleParams: [String]
-}
+scalar DateTime
+scalar JSONObject
 
-type SearchMapType {
-    _id: String
-    appId: String
-    apiConfig: ApiConfigType
-    fields: [FieldType]
-    preferences: SearchMapPreferenceType
-    createdBy: String
-    createdOn: DateTime
-    updatedBy: String
-    updatedOn: DateTime
+type SearchMap {
+  _id: String
+  id: String
```

Comment:
Do both id's serve the same purpose?

Reply:
Yeah, the field was `_id` before, I changed it to `id`. I think I missed it when resolving some conflicts.
Removed the `_id`.